src/main/java/uk/gov/companieshouse/controller/AppealController.java

```diff
@@ -0,0 +1,111 @@
+
+    private final AppealService appealService;
+
+    public AppealController(AppealService appealService) {
+        this.appealService = appealService;
+    }
+
+    @Operation(summary = "Create a new appeal", tags = "Appeal")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "201", description = "Appeal resource created", headers = {
+            @Header(name = "location")
+        }),
+        @ApiResponse(responseCode = "400", description = "Bad request"),
+        @ApiResponse(responseCode = "401", description = "Unauthorised request"),
+        @ApiResponse(responseCode = "422", description = "Invalid appeal data"),
+        @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
+    @PostMapping(value = "/{company-id}/appeals", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> submitAppeal(@RequestHeader("ERIC-identity") String userId,
+                                               @PathVariable("company-id") final String companyId,
+                                               @Valid @RequestBody final Appeal appeal) {
+
+        final String penaltyReference = appeal.getPenaltyIdentifier().getPenaltyReference();
+
+        LOGGER.info("POST /companies/{}/appeals with user id {} and penalty reference {}",
+            companyId, userId, penaltyReference);
+
+        if (StringUtils.isBlank(userId)) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+
+        try {
+            String id = appealService.saveAppeal(appeal, userId);
+
+            URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .path("/{id}")
+                .buildAndExpand(id)
+                .toUri();
+
+            return ResponseEntity.created(location).build();
+
+        } catch (Exception ex) {
+            LOGGER.error("Unable to create appeal for company number {}, penalty reference {} and user id {}",
+                companyId, penaltyReference, userId, ex);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @Operation(summary = "Get an appeal by ID", tags = "Appeal")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Appeal resource retrieved successfully"),
+        @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
+    @GetMapping(value = "/{company-id}/appeals/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Optional<Appeal>> getAppealById(@PathVariable("company-id") final String companyId,
+                                                          @PathVariable("id") final String id) {
+
+        LOGGER.info("GET /companies/{}/appeals/{}", companyId, id);
+
+        return ResponseEntity.status(HttpStatus.OK).body(appealService.getAppeal(id));
```

Comment:
That method should return 404 when no Appel is found, shouldn't it?

Reply:
Priority: must change.

Reply:
Sorry, i removed this as i thought we wanted to remove an optional empty. I'll put the code back - well part of it

Reply:
What I'd like to have is:
- service returns Optional
- controller returns 200 or 404 depending whether Optional is empty or not.

Reply:
Done :-)